include/puara/descriptors/touch.h

```diff
@@ -67,28 +64,15 @@ class Touch
     // normalized between 0 and 1
     touchBottom = touchAverage(discrete_touch, (touchSize - touchSizeEdge), touchSize);
 
-    // Save last blob detection state before reading new data
-    for(int i = 0; i < (sizeof(blobPos) / sizeof(blobPos[0])); ++i)
-    {
-      lastState_blobPos[i] = blobPos[i];
-    }
-
-    // 1D blob detection: used for brush
-    blobDetection1D(discrete_touch, touchSize);
+     // 1D blob detection: used for brush
+    const auto movement = blobDetector.detect1D(discrete_touch, touchSize);
 
     // brush: direction and intensity of capsense brush motion
     // rub: intensity of rub motion
     // in ~cm/s (distance between stripes = ~1.5cm)
-    for(int i = 0; i < (sizeof(blobPos) / sizeof(blobPos[0])); ++i)
+    for(int i = 0; i < maxNumBlobs; ++i)
```

Comment:
I think we can remove the iteration and simplify the class to process a single DoF